From bd6ea9a9a1da72e721cbcc3a3f23cc8c62675fce Mon Sep 17 00:00:00 2001
From: cristian64 <cristian64@gmail.com>
Date: Wed, 10 Dec 2025 21:11:08 +0000
Subject: [PATCH 2/2] Core/HW: Enable BBA (IPC) in FreeBSD.

Since v1.4.0, cpp-ipc now supports FreeBSD. This was a limitation that
prevented us from enabling compilation on FreeBSD in #13870.

Full changelog: https://github.com/mutouyun/cpp-ipc/releases/tag/v1.4.0
---
 CMakeLists.txt                                  | 2 +-
 Source/Core/Core/CMakeLists.txt                 | 2 +-
 Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h    | 4 ++--
 Source/Core/DolphinQt/Settings/GameCubePane.cpp | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2fa3769f8fde..0cadc9dd49e0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -786,7 +786,7 @@ endif()
 
 add_subdirectory(Externals/watcher)
 
-if(WIN32 OR LINUX)
+if(NOT ANDROID AND NOT APPLE)
   add_subdirectory(Externals/cpp-ipc)
 endif()
 
diff --git a/Source/Core/Core/CMakeLists.txt b/Source/Core/Core/CMakeLists.txt
index be75ba89d953..4f5e313ce748 100644
--- a/Source/Core/Core/CMakeLists.txt
+++ b/Source/Core/Core/CMakeLists.txt
@@ -801,7 +801,7 @@ if(UNIX)
   )
 endif()
 
-if(WIN32 OR LINUX)
+if(NOT ANDROID AND NOT APPLE)
   target_sources(core PRIVATE HW/EXI/BBA/IPC.cpp)
   target_link_libraries(core PRIVATE cpp-ipc::ipc)
 endif()
diff --git a/Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h b/Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h
index 67a03178312e..a6b078088655 100644
--- a/Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h
+++ b/Source/Core/Core/HW/EXI/EXI_DeviceEthernet.h
@@ -15,7 +15,7 @@
 #endif
 
 #include <SFML/Network.hpp>
-#if defined(WIN32) || (defined(__linux__) && !defined(__ANDROID__))
+#if !defined(__ANDROID__) && !defined(__APPLE__)
 #include <libipc/ipc.h>
 #endif
 
@@ -483,7 +483,7 @@ class CEXIETHERNET : public IEXIDevice
   public:
     explicit IPCBBAInterface(CEXIETHERNET* const eth_ref) : NetworkInterface(eth_ref) {}
 
-#if defined(WIN32) || (defined(__linux__) && !defined(__ANDROID__))
+#if !defined(__ANDROID__) && !defined(__APPLE__)
 
     bool Activate() override;
     void Deactivate() override;
diff --git a/Source/Core/DolphinQt/Settings/GameCubePane.cpp b/Source/Core/DolphinQt/Settings/GameCubePane.cpp
index 98460f6be07e..fd27612dc550 100644
--- a/Source/Core/DolphinQt/Settings/GameCubePane.cpp
+++ b/Source/Core/DolphinQt/Settings/GameCubePane.cpp
@@ -143,7 +143,7 @@ void GameCubePane::CreateWidgets()
            EXIDeviceType::EthernetXLink,
            EXIDeviceType::EthernetTapServer,
            EXIDeviceType::EthernetBuiltIn,
-#if defined(WIN32) || (defined(__linux__) && !defined(__ANDROID__))
+#if !defined(__APPLE__)
            EXIDeviceType::EthernetIPC,
 #endif
            EXIDeviceType::ModemTapServer,
